templates/vue/src/components/Lightbox/media/ActivityMedia/Question.vue

```diff
@@ -0,0 +1,414 @@
+        </div>
+        <div v-else>
+          <p>You haven't done the previous activity yet.</p>
+        </div>
+      </div>
+      <h1 class="question-title">
+        {{ question.text }}
+      </h1>
+      <b-alert
+        v-if="!isLoggedIn"
+        show
+        variant="warning"
+        class="loggedout-alert mx-auto"
+      >
+        Please login to have your answers saved.
+      </b-alert>
+      <div class="question-body">
+        <div v-if="formOpened">
+          <text-question
+            v-if="formType === 'text'"
+            :question="question"
+            :answer="answer"
+            @submit="handleSubmit"
+          ></text-question>
+          <audio-recorder
+            v-else-if="formType === 'audio'"
+            :id="question.id"
+            :node="node"
+            @submit="handleSubmit"
+          />
+        </div>
+        <div v-else class="question-answer-types">
+          <p class="question-answer-text">I want to answer with...</p>
+          <div class="button-container">
+            <answer-button
+              v-if="question.answerTypes.text.enabled"
+              :completed="textFormCompleted"
+              data-qa="answer-button-text"
+              @click="openForm('text')"
+            >
+              text
+            </answer-button>
+            <answer-button
+              v-if="question.answerTypes.audio.enabled"
+              :completed="audioFormCompleted"
+              icon="microphone"
+              data-qa="answer-button-audio"
+              @click="openForm('audio')"
+            >
+              audio
+            </answer-button>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { mapActions, mapGetters, mapState } from "vuex"
+import client from "@/services/TapestryAPI"
+import AnswerButton from "./AnswerButton"
+import AudioRecorder from "./AudioRecorder"
+import TextQuestion from "./TextQuestion"
+import Loading from "@/components/common/Loading"
+import TapestryActivity from "./TapestryActivity"
+import * as wp from "@/services/wp"
+import { data as wpData } from "@/services/wp"
+
+export default {
+  name: "question",
+  components: {
+    AnswerButton,
+    AudioRecorder,
+    TextQuestion,
+    Loading,
+    TapestryActivity,
+  },
+  props: {
+    question: {
+      type: Object,
+      required: true,
+    },
+    node: {
+      type: Object,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      submitting: false,
+      formOpened: false,
+      formType: "",
+      answers: {},
+    }
+  },
+  computed: {
+    ...mapGetters(["getAnswers", "getQuestion"]),
+    ...mapState(["userAnswers"]),
+    isLoggedIn() {
+      return wp.isLoggedIn()
+    },
+    lastQuestion() {
+      if (this.question.followUp.questionId !== null) {
+        return this.getQuestion(this.question.followUp.questionId)
+      }
+      return this.getQuestion(this.question.followUp.questionId)
```

Comment:
This is a moot if statement?

Reply:
@wynnset  I don't think this function is ever used
`lastQuestion() {
      if (this.question.followUp.questionId !== null) {
        return this.getQuestion(this.question.followUp.questionId)
      }
      return this.getQuestion(this.question.followUp.questionId)
    }`

Reply:
In that case why is it still here?